app/ts/library/ethereum.ts

```diff
@@ -0,0 +1,22 @@
+import { BigNumber, ethers } from 'ethers'
+import { TransactionReceipt, TransactionResponse, TransferTransactionResponse } from '../types.js'
+import { ERC20ABI } from './ERC20ABI.js'
+
+export function isTransferTransaction(txResponse: TransactionResponse): txResponse is TransferTransactionResponse {
+	return txResponse.data.startsWith('0xa9059cbb')
+}
+
+export function getTransferTokenValue(transactionReceipt: TransactionReceipt): BigNumber {
+	const IERC20 = new ethers.utils.Interface(ERC20ABI)
+
+	let tokenValue = undefined
+
+	for (const receiptLog of transactionReceipt.logs) {
+		const parsedLog = IERC20.parseLog(receiptLog)
```

Comment:
What happens if `parseLog` fails?  Does it throw or return something else?  It is very possible that there are multiple events in a transfer, some of which may not be parsable with `ERC20ABI`.

Reply:
Adjusted in
https://github.com/DarkFlorist/lunaria/pull/89/commits/540795b94176d74702c283d631cf134bc1aa761a. Now it should return `undefined` to hand over handling to the next call